ReactAndroid/src/main/java/com/facebook/react/uimanager/NativeViewHierarchyOptimizer.java

```diff
@@ -435,15 +439,39 @@ private void transitionLayoutOnlyViewToNativeView(
     mTagsWithLayoutVisited.clear();
   }
 
-  private static boolean isLayoutOnlyAndCollapsable(@Nullable ReactStylesDiffMap props) {
+  private static boolean isCollapsableChildren(@Nullable ReactStylesDiffMap props) {
```

Comment:
I think we can improve the name here... it's a bit confusing to me. Maybe `areChildrenCollapsable`?